sklearn/utils/testing.py

```diff
@@ -388,14 +388,9 @@ def _assert_allclose(actual, desired, rtol=1e-7, atol=0,
 
 
 def assert_raise_message(exception, message, function, *args, **kwargs):
-    """Helper function to test error messages in exceptions"""
-
-    try:
-        function(*args, **kwargs)
-        raise AssertionError("Should have raised %r" % exception(message))
-    except exception as e:
-        error_message = str(e)
-        assert_in(message, error_message)
+    """Calls assert_raises_regex with message as regular expression"""
```

Comment:
I think the docstring should be more end-user friendly:

```
    """Check that exception is raised with the expected message"""
```

The fact that we are delegating the check to the more general `assert_raises_regex` is an implementation detail.